stdlib/asm/mem.masm

```diff
@@ -141,30 +144,30 @@ export.pipe_words_to_memory.0
     # => [B', A, write_ptr+1, ...]
 
     # Push padding word (4 cycles)
-    push.1.0.0.0
+    padw
     # => [C, B', A, write_ptr+1, ...]
 
     # Run RPO permutation (1 cycles)
     hperm
     # => [C', B', A', write_ptr+1, ...]
   end
-
-  # The RPO result is word B, discard the unused portion of the rate and the capacity. (9 cycles)
-  dropw swapw dropw
-  # => [rpo_hash, write_ptr', ...]
 end
 
 #! Moves an arbitrary number of words from the advice stack to memory and asserts it matches the commitment.
 #!
 #! Input: [num_words, write_ptr, COM, ...]
 #! Output: [write_ptr', ...]
 #! Cycles:
-#!  even num_words: 58 + 9 * num_words / 2
-#!  odd num_words: 75 + 9 * round_down(num_words / 2)
+#!  even num_words: 67 + 9 * num_words / 2
+#!  odd num_words: 84 + 9 * round_down(num_words / 2)
 export.pipe_preimage_to_memory.0
```

Comment:
Why did the cycle count increase by this much? We've added a call to `rpo::squeeze_digest` (which added 9 cycles) but `pipe_words_to_memory` should now be simpler, no? So, the delta should be about 2 cycles?

Reply:
The count is wrong. Here is a second attempt at it:
- `pipe_words_to_memory`: 41 + 9 * num_words / 2
- `squeeze_digest`: 9
- `movup.4 movdn.8`: 2
- `assert_eqw`: 10

The total is `62 + 9 * num_words / 2` for even number of words.
Similarly, the total is `79 + 9 * num_words / 2` for odd number of words.